jpo-ode-svcs/src/test/java/us/dot/its/jpo/ode/traveler/TimDepositControllerTest.java

```diff
@@ -118,32 +117,44 @@ public void failedXmlConversionShouldReturnConversionError(@Capturing TimTransmo
          {
             TimTransmogrifier.convertToXml((DdsAdvisorySituationData) any, (ObjectNode) any, (OdeMsgMetadata) any,
                   (SerialId) any);
-            result = new  XmlUtilsException("testException123", null);
+            result = new XmlUtilsException("testException123", null);
          }
       };
 
       ResponseEntity<String> actualResponse = testTimDepositController.postTim(
             "{\"request\":{\"rsus\":[],\"snmp\":{}},\"tim\":{\"msgCnt\":\"13\",\"timeStamp\":\"2017-03-13T01:07:11-05:00\"}}");
-      assertEquals("{\"error\":\"Error sending data to ASN.1 Encoder module: testException123\"}", actualResponse.getBody());
+      assertEquals("{\"error\":\"Error sending data to ASN.1 Encoder module: testException123\"}",
+            actualResponse.getBody());
 
    }
-   
+
    @Test
-   public void testSuccessfulMessageReturnsSuccessMessagePost(@Capturing XmlUtils capturingXmlUtils) {
+   public void testSuccessfulMessageReturnsSuccessMessagePost(@Capturing TimTransmogrifier capturingTimTransmogrifier, @Capturing XmlUtils capturingXmlUtils) {
       ResponseEntity<String> actualResponse = testTimDepositController.postTim(
             "{\"request\":{\"rsus\":[],\"snmp\":{}},\"tim\":{\"msgCnt\":\"13\",\"timeStamp\":\"2017-03-13T01:07:11-05:00\"}}");
       assertEquals("{\"success\":\"true\"}", actualResponse.getBody());
    }
-   
+
    @Test
-   public void testSuccessfulMessageReturnsSuccessMessagePostWithOde(@Capturing XmlUtils capturingXmlUtils) {
+   public void testSuccessfullSdwRequestMessageReturnsSuccessMessagePost()
+         throws Exception {
+      String file = "/sdwRequest.json";
+      String json = IOUtils.toString(
+            TimDepositControllerTest.class.getResourceAsStream(file),
+            "UTF-8");
+      ResponseEntity<String> actualResponse = testTimDepositController.postTim(json);
```

Comment:
Do we need to run the application in order for this unit test to pass?

Reply:
Is this test not running for you? All tests pass within the development environment for me. I left out the TimTransmogrifier and XmlUtils capture here to see that a full TIM JSON actually passes though the process successfully. Previously this would show a success even if the TimTransmogrifier or XmlUtils portion produced an invalid value. This was causing difficulties in detecting the invalid JSON issues from the Jackson library. If you'd like different tests here, please let me know.

Reply:
Hello @payneBrandon Sorry for the delay response. 
This is the error messages when I run the unit test.
```
[INFO] Running us.dot.its.jpo.ode.traveler.TimDepositControllerTest
us.dot.its.jpo.ode.util.JsonUtils$JsonUtilsException: Error deserializing JSON tree to us.dot.its.jpo.ode.model.OdeTravelerInputData
	at us.dot.its.jpo.ode.util.JsonUtils.jacksonFromJson(JsonUtils.java:95)
	at us.dot.its.jpo.ode.util.JsonUtils.fromJson(JsonUtils.java:84)
	at us.dot.its.jpo.ode.traveler.TimDepositController.depositTim(TimDepositController.java:123)
	at us.dot.its.jpo.ode.traveler.TimDepositController.postTim(TimDepositController.java:280)
	at us.dot.its.jpo.ode.traveler.TimDepositControllerTest.invalidJsonSyntaxShouldReturnJsonSyntaxError(TimDepositControllerTest.java:66)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at mockit.integration.junit4.JUnit4TestRunnerDecorator.executeTestMethod(JUnit4TestRunnerDecorator.java:157)
	at mockit.integration.junit4.JUnit4TestRunnerDecorator.invokeExplosively(JUnit4TestRunnerDecorator.java:71)
	at mockit.integration.junit4.FakeFrameworkMethod.invokeExplosively(FakeFrameworkMethod.java:29)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:157)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "invalid" (class us.dot.its.jpo.ode.model.OdeTravelerInputData), not marked as ignorable (2 known properties: "tim", "request"])
 at [Source: (String)"{"invalid":"json"}}"; line: 1, column: 13] (through reference chain: us.dot.its.jpo.ode.model.OdeTravelerInputData["invalid"])
	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:987)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1974)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1701)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1679)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:330)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:187)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:322)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4593)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3548)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3516)
	at us.dot.its.jpo.ode.util.JsonUtils.jacksonFromJson(JsonUtils.java:93)
	... 47 more
```

Reply:
Hey @dan-du-car, I'll look into this a bit more, but currently running the test manually or with a `mvn test` command both seem to work fine on my end. When you debug the test, if you put a breakpoint on line 145 before calling the postTim method, is the "json" string value populated properly? It should be reading in the sdwRequest.json file from resources.

Reply:
The json at line 142 is fine. I printed out the json content, it is the same as in the sdwRequest.json file.

Reply:
Ok, that rules out file loading problems. How are you running the test? What is your development environment?

Reply:
It seems from the invalidJsonSyntaxShouldReturnJsonSyntaxError() test case. The exception is not caught.

Reply:
It seems to be from here. https://github.com/usdot-jpo-ode/jpo-ode/blob/97faaaa2cad9773b47e5f2be1af758c906c03b51/jpo-ode-common/src/main/java/us/dot/its/jpo/ode/util/JsonUtils.java#L86
The exception is caught, and printed out the stack trace messages.